### PR TITLE
Allow installing Node.js version based on file

### PR DESCRIPTION
**Pull request by ismayilov-ismayil**
**Task name**: Allow installing Node.js version based on .nvmrc

**Description**: Currently we have only option to specify Node version during pipeline creation. We add new option to get version from .nvmrc file

**Before**

**After**

**Documentation changes required:** (Y/N) Y

**Added unit tests:** (Y/N) Y

**Attached related issue:** (Y/N)  Y
[https://github.com/microsoft/azure-pipelines-tasks/issues/14094](https://github.com/microsoft/azure-pipelines-tasks/issues/14094)
**Checklist**:
- [x] Task version was bumped - v0.210.1
- [x] Checked that applied changes work as expected

### CODE REVIEW

**Review by vmapetr — APPROVED** [review]

**Review by DmitriiBobreshev — APPROVED** [review]

File: Tasks/NodeToolV0/Strings/resources.resjson/en-US/resources.resjson [hunk 1]
```diff
@@ -5,6 +5,8 @@
   "loc.instanceNameFormat": "Use Node $(versionSpec)",
   "loc.input.label.versionSpec": "Version Spec",
   "loc.input.help.versionSpec": "Version Spec of version to get.  Examples: 6.x, 4.x, 6.10.0, >=6.10.0",
+  "loc.input.label.versionFilePath": "Path of file",
```
Comment (max-zaytsev) [inline Tasks/NodeToolV0/Strings/resources.resjson/en-US/resources.resjson:8]: I would call it "Path to the .nvmrc file"

File: Tasks/NodeToolV0/Strings/resources.resjson/en-US/resources.resjson [hunk 2]
```diff
@@ -5,6 +5,8 @@
...
+  "loc.input.help.versionFilePath": "File path to get version.  Examples: src/.nvmrc",
```
Comment (max-zaytsev) [inline Tasks/NodeToolV0/Strings/resources.resjson/en-US/resources.resjson:9]:
```suggestion
  "loc.input.help.versionFilePath": "File path to get version.  Example: src/.nvmrc",
```

File: Tasks/NodeToolV0/Strings/resources.resjson/en-US/resources.resjson (hunk 1 above)
Comment (ismayilov-ismayil) [reply Tasks/NodeToolV0/Strings/resources.resjson/en-US/resources.resjson:8]: > I would call it "Path to the .nvmrc file"

Actually, I want to keep it not depend on file extension, so any file extension can be used to provide Node version. I mean, we can use 'nodeversion.txt' file as well

Comment (max-zaytsev) [reply Tasks/NodeToolV0/Strings/resources.resjson/en-US/resources.resjson:8]: Do we really need this :) ? Anyway, I would rephrase it in a more descriptive way since you need to see help message to understand what type of file is needed

**Review by kirill-ivlev — APPROVED** [review]

**Review by max-zaytsev — APPROVED** [review]

**Review by wojgasior — APPROVED** [review]

**Comment by atwright147** [comment]
Hey, I cannot see this option in my pipeline, how do I get to take advantage of this?

And how would I use this when writing in YAML?

Sorry for adding noise to this PR
